pandas-stubs/core/frame.pyi

```diff
@@ -919,28 +922,28 @@ class DataFrame(NDFrame, OpsMixin):
     ) -> DataFrame: ...
     def diff(self, periods: int = ..., axis: AxisType = ...) -> DataFrame: ...
     @overload
-    def agg(self, func: Callable | _str, axis: AxisType = ..., **kwargs) -> Series: ...
+    def agg(self, func: AggFuncTypeBase, axis: AxisType = ..., **kwargs) -> Series: ...
     @overload
     def agg(
         self,
-        func: list[Callable] | dict[_str, Callable] = ...,
+        func: list[AggFuncTypeBase] | AggFuncTypeDict = ...,
```

Comment:
This and some of the other changes allow more types. Can you please add tests for these newly accepted types, for example `func: list[str]`, `func: dict[str, list[Callable]]`, and `func: dict[str, list[str]]`?